### PR TITLE
Add services' aliases to improve autowiring experience

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

### CODE REVIEW

File: src/Bundle/Resources/config/services.xml [hunk 1]
```diff
@@ -35,18 +35,22 @@
                  decorates="form.type_extension.form.request_handler" decoration-priority="256" public="false" />
 
         <service id="sylius.resource_registry" class="Sylius\Component\Resource\Metadata\Registry" public="false" />
+        <service id="Sylius\Component\Resource\Metadata\RegistryInterface" alias="sylius.resource_registry" />
```
Comment (loic425) [inline src/Bundle/Resources/config/services.xml]: public false is missing

File: src/Bundle/Resources/config/services/controller.xml [hunk 2]
```diff
@@ -19,40 +19,69 @@
...
         <service id="sylius.resource_controller.new_resource_factory" class="Sylius\Bundle\ResourceBundle\Controller\NewResourceFactory" public="false" />
+        <service id="Sylius\Bundle\ResourceBundle\Controller\NewResourceFactoryInterface" alias="sylius.resource_controller.new_resource_factory" public="false" />
+
         <service id="sylius.resource_controller.single_resource_provider" class="Sylius\Bundle\ResourceBundle\Controller\SingleResourceProvider" public="false" />
-        <service id="sylius.resource_controller.pagerfanta_representation_factory" class="Hateoas\Representation\Factory\PagerfantaFactory" public="false" />
+        <service id="Sylius\Bundle\ResourceBundle\Controller\SingleResourceProviderInterface" alias="sylius.resource_controller.single_resource_provider" public="false" />
+
+        <service id="Hateoas\Representation\Factory\PagerfantaFactory" public="false" />
+        <service id="sylius.resource_controller.pagerfanta_representation_factory" alias="Hateoas\Representation\Factory\PagerfantaFactory" public="false" />
+
```
Comment (Roshyo) [inline src/Bundle/Resources/config/services/controller.xml:39]: I am wondering. For consistency reasons, I'd rather go with named service, then alias instead of service, then named alias.

WDYT?

File: src/Bundle/Resources/config/services/form.xml [hunk 3]
```diff
@@ -6,9 +6,10 @@
     <services>
         <defaults public="true" />
 
-        <service id="sylius.form.type.resource_autocomplete_choice" class="Sylius\Bundle\ResourceBundle\Form\Type\ResourceAutocompleteChoiceType">
+        <service id="Sylius\Bundle\ResourceBundle\Form\Type\ResourceAutocompleteChoiceType">
             <argument type="service" id="sylius.registry.resource_repository" />
             <tag name="form.type" />
         </service>
+        <service id="sylius.form.type.resource_autocomplete_choice" alias="Sylius\Bundle\ResourceBundle\Form\Type\ResourceAutocompleteChoiceType" />
```
Comment (Roshyo) [inline src/Bundle/Resources/config/services/form.xml:13]: Is this change necessary? Autowire should already work with previous definition, shouldn't it?

File: src/Bundle/Resources/config/services/integrations/doctrine.xml [hunk 4]
```diff
@@ -15,11 +15,11 @@
...
 
-        <service id="sylius_resource.doctrine.mapping_driver_chain"
-                 class="Sylius\Bundle\ResourceBundle\Doctrine\ResourceMappingDriverChain"
+        <service id="Sylius\Bundle\ResourceBundle\Doctrine\ResourceMappingDriverChain"
                  decorates="doctrine.orm.default_metadata_driver">
-            <argument type="service" id="sylius_resource.doctrine.mapping_driver_chain.inner" />
+            <argument type="service" id="Sylius\Bundle\ResourceBundle\Doctrine\ResourceMappingDriverChain.inner" />
             <argument type="service" id="sylius.resource_registry" />
         </service>
+        <service id="sylius_resource.doctrine.mapping_driver_chain" alias="Sylius\Bundle\ResourceBundle\Doctrine\ResourceMappingDriverChain" />
```
Comment (Roshyo) [inline src/Bundle/Resources/config/services/integrations/doctrine.xml:23]: Same as form type above

File: src/Bundle/Resources/config/services/form.xml (hunk 3 above)
Comment (loic425) [reply src/Bundle/Resources/config/services/form.xml:13]: Indeed, this change is not for autowiring. new services in Sylius uses FQCN for service id or the interface. So it's to slightly move to FQCN ids.

**Review by Roshyo — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]

File: src/Bundle/Resources/config/services/controller.xml [hunk 5]
```diff
@@ -19,40 +19,69 @@
             <argument type="service" id="service_container" />
             <argument type="service" id="sylius.expression_language" />
         </service>
+        <service id="Sylius\Bundle\ResourceBundle\Controller\ParametersParserInterface" alias="sylius.resource_controller.parameters_parser" public="false" />
```
Comment (lchrusciel) [inline src/Bundle/Resources/config/services/controller.xml:22]: Can we reverse this definition? So the interface will be on a service itself and `sylius.resource_controller.parameters_parser` will alias to it? Same as requested: https://github.com/Sylius/SyliusGridBundle/pull/279#discussion_r1021603167

**Review by lchrusciel — APPROVED** [review]

File: src/Bundle/Resources/config/services/console.xml [hunk 6]
```diff
@@ -15,9 +15,10 @@
     <services>
         <defaults public="true" />
 
-        <service id="sylius.console.command.resource_debug" class="Sylius\Bundle\ResourceBundle\Command\DebugResourceCommand">
+        <service id="Sylius\Bundle\ResourceBundle\Command\DebugResourceCommand">
             <argument type="service" id="sylius.resource_registry" />
             <tag name="console.command" />
         </service>
+        <service id="sylius.console.command.resource_debug" alias="Sylius\Bundle\ResourceBundle\Command\DebugResourceCommand" />
```
Comment (mbabker) [inline src/Bundle/Resources/config/services/console.xml:22]: @loic425 @lchrusciel Changes like this one could technically be a B/C break.  You're changing a service ID from a definition to an alias.  Granted, with this specific case, it's probably not a big deal, but if someone were doing something in a compiler pass, a `$container->getDefinition('sylius.console.command.resource_debug')` call would no longer work.
